src/charm.py

```diff
@@ -380,6 +394,31 @@ def _set_status_gateway_address(
         else:
             self.unit.status = WaitingStatus("Gateway address unavailable")
 
+    def _certificates_revocation_needed(self, client: Client, config: CharmConfig) -> bool:
+        """Check if a new certificate is needed.
+
+        Args:
+            client: Lightkube client
+            config: Charm config.
+
+        Returns:
+            True if the current certificate needs to be revoked.
+        """
+        gateway_resource_manager = GatewayResourceManager(
+            labels=self._labels,
+            client=client,
+        )
+        gateway = gateway_resource_manager.current_gateway_resource()
+        if not gateway:
+            return False
+
+        gateway_listeners = gateway.spec["listeners"]
+        listener_hostnames = [listener["hostname"] for listener in gateway_listeners]
+        if len(set(listener_hostnames)) == 1 and config.external_hostname == listener_hostnames[0]:
```

Comment:
Can someone change the gateway-class and this returns True, revoking certificates even if is not needed?

Reply:
The method only returns true if the hostname in the listeners is different from what is configured, so i think it doesn't matter which config triggers the hook method if revocation is not needed then the charm should do nothing.

Reply:
Lets keep an eye on that to guarantee that a unrelated config-change triggers a certificate revocation.

Reply:
Yep, I'll keep that in mind.